### PR TITLE
feat: get reason of confirmed transaction

### PR DESCRIPTION
**Pull request by rodolfopietro97**
# Description

Should be able to get the revert reason of a transaction

### CODE REVIEW

File: packages/network/src/thor-client/transactions/transactions-module.ts [hunk 1]
```diff
@@ -298,6 +309,97 @@ class TransactionsModule {
...
+    }
+
+    /**
+     * Get the revert reason of an existing transaction.
+     *
+     * @param transactionHash - The hash of the transaction to get the revert reason for.
+     * @returns A promise that resolves to the revert reason of the transaction.
+     */
+    public async getRevertReason(
+        transactionHash: string
```
Comment (darrenvechain) [inline packages/network/src/thor-client/transactions/transactions-module.ts:347]: @rodolfopietro97 can we optionally add a error fragment to decode custom errors?

https://docs.ethers.org/v6/api/abi/abi-coder/#ErrorFragment

Comment (darrenvechain) [reply packages/network/src/thor-client/transactions/transactions-module.ts:347]: See here an example of a custom error: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L168

Comment (darrenvechain) [reply packages/network/src/thor-client/transactions/transactions-module.ts:347]: Actually, an array of errors might be better? That way I can pass in all of the errors that my contract have, and find the correct reason

File: packages/network/src/thor-client/transactions/transactions-module.ts [hunk 2]
```diff
@@ -298,6 +309,97 @@ class TransactionsModule {
...
+        const blocksModule = new BlocksModule(this.thor);
+        const debugModule = new DebugModule(this.thor);
+
+        // 2 - Get the transaction details
+        const transaction = (await this.getTransaction(
+            transactionHash
+        )) as TransactionDetailNoRaw;
+
+        // 3 - Get the block details (to get the transaction index)
+        const block = (await blocksModule.getBlockExpanded(
```
Comment (darrenvechain) [inline packages/network/src/thor-client/transactions/transactions-module.ts:359]: `this.thor.blocks` maybe? I think theres no need to create new modules

Comment (rodolfopietro97) [reply packages/network/src/thor-client/transactions/transactions-module.ts]: https://github.com/vechain/vechain-sdk-js/pull/862/commits/cdaa2951395c946190c8cbb63564ba22fdad057d

Oh yes I saw yesterday. Solved here

File: packages/network/src/thor-client/transactions/transactions-module.ts (hunk 1 above)
Comment (rodolfopietro97) [reply packages/network/src/thor-client/transactions/transactions-module.ts:347]: https://github.com/vechain/vechain-sdk-js/issues/866

**Review by fabiorigam — APPROVED** [review]

**Pull request merged**